shiny/ui/_sidebar.py

```diff
@@ -326,32 +337,61 @@ def layout_sidebar(
 
     See Also
```

Comment:
`layout_sidebar` should drop `sidebar` parameter. It should inspect `*args` for the Sidebar object.

Reply:
I'd prefer to see `sidebar` as an (optional) keyword parameter, similar in spirit to what we have on the R side:

```r
layout_sidebar <- function(..., sidebar = NULL) {}
``` 

```python
def layout_sidebar(*args, sidebar = None):
  pass
```

I'm aware that this implies that the sidebar definition would have to come after the main content (e.g., `layout_sidebar("Main", sidebar = "Side")`, but here's some considerations that make me feel OK about that tradeoff:

1. It doesn't feel obviously wrong to me for the main content to come first.
2. Given there is a preference for sidebar first, this is fits into a class of problem that isn't isolated to `layout_sidebar()`. `page_*()` functions (e.g., `title`, etc) and anything else that wants to prioritize a collection of UI elements 1st and (optional) "parent" parameters 2nd (e.g., `layout_column_wrap(card(), card(), width=200)`) will also suffer from this problem, so it at least fits an "existing pattern" that'll hopefully feel familiar already.
3. With your suggestion (inspecting `*args`), it seems like it'd be relatively easy for a user to wrap a `sidebar()` in some other object and encounter an error or undefined behavior.
4. With your suggestion (inspecting `*args`), it becomes harder to align the R<->Python APIs. It's not obvious to me how we'd add support for this usage on R side (e.g., if we were to inspect `...` for a `sidebar()`, what should happen if a `sidebar` argument is also used?). That might be made easier by making a breaking change, but I'd _really_ hate to do that in order to fix something that isn't really a problem in R.

Also, more generally, if and when API changes are being proposed when porting R -> Python, @gadenbuie and I would appreciate being involved in those discussions/decisions.

Reply:
Addressing 3. ... the dev can return a `Sidebar` class.
Addressing 4. ... I don't believe bslib needs to change their API. There can be deviations. It is ok if they do not have the same exact interface for everything. There are fundamental difference in how we can express arguments in the two languages. Ex: The split between `TagAttrs` in an `*args` and `TagAttrValue` in `**kwargs` vs just `...` in R.

-----------------------

You and I talked about it yesterday and we couldn't decide on a straight path forward. I brought all motivations we had for the different approaches in conversation with @wch that afternoon. 

We'll discuss as a team about this method and a handful of others on Friday.

------------------------

In the interest of moving forward, I'm continuing with the current API and merging the PR.